daemon/container_unix.go

```diff
@@ -948,7 +948,8 @@ func (container *Container) connectToNetwork(idOrName string, updateSettings boo
 
 	ep, err := container.getEndpointInNetwork(n)
 	if err == nil {
-		return fmt.Errorf("container already connected to network %s", idOrName)
```

Comment:
I dont think hiding this error is appropriate.
There can be genuine cases such as https://github.com/docker/docker/issues/15995 where race conditions can result in a genuine case of multiple containers trying to use the same endpoint resulting in this error.